packages/react-server/src/ReactFlightServer.js

```diff
@@ -705,6 +705,11 @@ function serializeThenable(
         // to do so again here.
         erroredTask(request, newTask, reason);
         enqueueFlush(request);
+
+        if (request.abortableTasks.size === 0) {
+          const onAllReady = request.onAllReady;
+          onAllReady();
+        }
```

Comment:
The bug mentioned also affects prerenders where the last task to complete is a thenable that rejects right? because in performWork the onAllReady would never be called and the rejection would lead to the erroredTask call above which needs to hit this special case where onAllReady is called to work right?

There are actually a number of places where this same flaw likely exists like with serializing streams.

It seems like maybe we ought to move the onAllReady call into retryTask, erroredTask, and abort so that we only ever call it once